.github/workflows/test-e2e-cypress.yml

```diff
@@ -28,7 +28,163 @@ on:
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Setup Spec Matrix
+        id: setup-spec-matrix
+        run: |
+          changed_files=$(git diff --name-only --diff-filter=ACDMRTUXB ${{ github.event.before }} origin/master)
+          spec_list=()
+
+          for line in $changed_files; do
+            if [[ "$line" =~ src/(blocks|extensions|componets).* ]]; then # If changed file is within our tested paths.
+
+              changed_path=$(echo $line | cut -d'/' -f3) # Get the main path of the changed file, e.g. author, food-and-drinks, lightbox.
+
+              found_specs=$(find . -type f -name "*.cypress.js") # Find real spec files.
+              pattern="/src/(blocks|extensions|components)/${changed_path}/" # Changed files pattern to match against.
+
+              for spec in $found_specs; do
+                if [[ $spec  =~ $pattern ]]; then
+
+                  # Remove the path from the spec.
+                  spec=$(echo $spec | sed 's:.*/::')
+
+                  # Build spec list for cypress run.
+                  spec_list+="'${spec}',"
+
+                fi
+              done
+
+            fi
+          done
+
+          # All changes processed. Format string for cypress run.
+          # Remove trailing comma from spec_list.
+          updated_string=$(echo $spec_list | sed 's/,$//' )
+
+          # Wrap list in brackets for GH workflow matrix.
+          wrapped_string="[$updated_string]"
+
+          # Save the matrix array to the output.
+          echo "Testing against the following specs:"
+          echo $wrapped_string
+          echo "matrix=$wrapped_string" >> "$GITHUB_OUTPUT"
+
+  test_cypress_e2e_changed_files:
+    needs: setup-matrix
+    if: github.event_name == 'pull_request' && needs.setup-matrix.outputs.matrix != '[]'
+    name: E2E Test - Changed Files
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        spec: ${{ fromJson(needs.setup-matrix.outputs.matrix) }}
+
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Setup wp-env
+        uses: godaddy-wordpress/setup-wp-env@v1
+        with:
+          core: ${{ inputs.wpVersion }}
+          phpVersion: ${{ inputs.phpVersion }}
+          plugins: '["."]'
+          themes: '["https://downloads.wordpress.org/theme/go.zip"]'
+
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version-file: '.nvmrc'
+          cache: 'yarn'
+
+      - name: Install dependencies
+        run: |
+          composer install --prefer-dist --optimize-autoloader &
+          yarn install --immutable
+
+      - name: Build plugin
+        run: |
+          npx grunt build
+          # moving the built directory to this location means no dev files in tests-container.
+          mv ./build/coblocks $(wp-env install-path)/${{ inputs.installPath }}/wp-content/plugins/
```

Comment:
Can we move this into it's own job so we can share the built plugin files with all the runs in this matrix? It takes 2:00-2:30 min to complete just these two steps.

Reply:
I think that is a good idea. From my testing with the dependencies, uploading the files then re-downloading the artifact in a new job took the same if not longer and was not an efficient difference. However, I believe that building the plugin and needing to re-download just the compiled and minified scripts should help these workflow runtimes for sure.

Reply:
Hmm, ok maybe not worth it without more optimizations then. Removing the node_modules directory would speed it up but we need those dependencies to run the tests too. This can be ignored for now.